sql3/planner/oppredict.go

```diff
@@ -0,0 +1,253 @@
+			ColumnName:   fmt.Sprintf("predicted_%s", labelName),
+			RelationName: "",
+			AliasName:    "",
+			// we need to get this type from somewhere...probably needs to be stored in the model def
+			Type: &parser.DataTypeDecimal{
+				Scale: 4,
+			},
+		})
+
+	default:
+		// don't add anything
+	}
+	// add the columns from the select
+	result = append(result, p.ChildOp.Schema()...)
+
+	return result
+}
+
+func (p *PlanOpPredict) Iterator(ctx context.Context, row types.Row) (types.RowIterator, error) {
+	// get the query iterator
+	iter, err := p.ChildOp.Iterator(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	switch strings.ToLower(p.model.modelType) {
+	case "linear_regresssion":
+		return newLinearRegressionPredictIter(p.planner, p.model, p.ChildOp.Schema(), iter), nil
+
+	default:
+		return nil, sql3.NewErrInternalf("unexpected model tyoe '%s'", p.model.modelType)
+	}
+}
+
+func (p *PlanOpPredict) Children() []types.PlanOperator {
+	return []types.PlanOperator{
+		p.ChildOp,
+	}
+}
+
+func (p *PlanOpPredict) WithChildren(children ...types.PlanOperator) (types.PlanOperator, error) {
+	if len(children) != 1 {
+		return nil, sql3.NewErrInternalf("unexpected number of children '%d'", len(children))
+	}
+	return NewPlanOpPredict(p.planner, p.model, children[0]), nil
+}
+
+func (p *PlanOpPredict) Plan() map[string]interface{} {
+	result := make(map[string]interface{})
+	result["_op"] = fmt.Sprintf("%T", p)
+	sc := make([]string, 0)
+	for _, e := range p.Schema() {
+		sc = append(sc, fmt.Sprintf("'%s', '%s', '%s'", e.ColumnName, e.RelationName, e.Type.TypeDescription()))
+	}
+	result["_schema"] = sc
+	result["model"] = p.model.name
+	result["child"] = p.ChildOp.Plan()
+	return result
+}
+
+func (p *PlanOpPredict) String() string {
+	return ""
+}
+
+func (p *PlanOpPredict) AddWarning(warning string) {
+	p.warnings = append(p.warnings, warning)
+}
+
+func (p *PlanOpPredict) Warnings() []string {
+	var w []string
+	w = append(w, p.warnings...)
+	if p.ChildOp != nil {
+		w = append(w, p.ChildOp.Warnings()...)
+	}
+	return w
+}
+
+type linearRegressionPredictIter struct {
+	child         types.RowIterator
+	planner       *ExecutionPlanner
+	model         *modelSystemObject
+	regres        *regression.Regression
+	childSchema   types.Schema
+	inferenceRefs []*qualifiedRefPlanExpression
+	hasStarted    *struct{}
+}
+
+func newLinearRegressionPredictIter(planner *ExecutionPlanner, model *modelSystemObject, childSchema types.Schema, child types.RowIterator) *linearRegressionPredictIter {
+	return &linearRegressionPredictIter{
+		planner:       planner,
+		model:         model,
+		child:         child,
+		childSchema:   childSchema,
+		regres:        new(regression.Regression),
+		inferenceRefs: make([]*qualifiedRefPlanExpression, 0),
+	}
+}
+
+func (i *linearRegressionPredictIter) Next(ctx context.Context) (types.Row, error) {
+	if i.hasStarted == nil {
+		// label column
+		i.regres.SetObserved("Murders per annum per 1,000,000 inhabitants")
```

Comment:
i think i was expecting the linear regression to be more user-defined and less hard-coded?

Reply:
Yeah...it should pull this out of the schema...